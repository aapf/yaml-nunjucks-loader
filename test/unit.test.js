const path = require('path')
const { compile, sleep } = require('./helpers')

jest.setTimeout(5000)

describe('[unit] loader', () => {
  afterAll(async () => {
    await sleep(1000) // wait until webpack releases handles (avoid jest warning)
  })

  test('works with sample1.js', async () => {
    const sample1Path = path.resolve(__dirname, 'sample1.js')
    const modules = await compile(sample1Path)

    const js = modules.find((m) => m.name.endsWith('sample1.js'))
    expect(js).toBeTruthy()
    expect(js.source).toMatch(/var sample1 = require\('.\/sample1\.yaml'\);/)

    const yaml = modules.find((m) => m.name.endsWith('sample1.yaml'))
    expect(yaml).toBeTruthy()
    expect(yaml.source).toMatch(/var nunjucks = require\(".*nunjucks\/browser\/nunjucks-slim.*"\);/)
    expect(yaml.source).toMatch(/var YAML = require\(".*yaml.*"\);/)
    expect(yaml.source).toMatch(/\{"autoescape":false\}/)
    expect(yaml.source).toMatch(/return YAML\.parse\(env\.render\("sample1\.yaml", params\)\);/)
    expect(yaml.source).toMatch(/module\.exports = render;/)
    expect(yaml.source).toMatchSnapshot()
  })

  test('works with sample2.js', async () => {
    const sample2Path = path.resolve(__dirname, 'sample2.js')
    const modules = await compile(sample2Path)

    const js = modules.find((m) => m.name.endsWith('sample2.js'))
    expect(js).toBeTruthy()
    expect(js.source).toMatch(/const sample2 = require\('.\/sample2\.yaml'\);/)

    const yaml = modules.find((m) => m.name.endsWith('sample2.yaml'))
    expect(yaml).toBeTruthy()
    expect(yaml.source).toMatch(/var nunjucks = require\(".*nunjucks\/browser\/nunjucks-slim.*"\);/)
    expect(yaml.source).toMatch(/var YAML = require\(".*yaml.*"\);/)
    expect(yaml.source).toMatch(/\{"autoescape":false\}/)
    expect(yaml.source).toMatch(/return YAML\.parse\(env\.render\("sample2\.yaml", params\)\);/)
    expect(yaml.source).toMatch(/module\.exports = render;/)
    expect(yaml.source).toMatchSnapshot()
  })

  test('works with loader options', async () => {
    const sample2Path = path.resolve(__dirname, 'sample2.js')
    const modules = await compile(sample2Path, 'web', 'development', true, {
      nunjucks: { autoescape: true, throwOnUndefined: true }
    })

    const js = modules.find((m) => m.name.endsWith('sample2.js'))
    expect(js).toBeTruthy()
    expect(js.source).toMatch(/const sample2 = require\('.\/sample2\.yaml'\);/)

    const yaml = modules.find((m) => m.name.endsWith('sample2.yaml'))
    expect(yaml).toBeTruthy()
    expect(yaml.source).toMatch(/var nunjucks = require\(".*nunjucks\/browser\/nunjucks-slim.*"\);/)
    expect(yaml.source).toMatch(/var YAML = require\(".*yaml.*"\);/)
    expect(yaml.source).toMatch(/\{"autoescape":true,"throwOnUndefined":true\}/)
    expect(yaml.source).toMatch(/return YAML\.parse\(env\.render\("sample2\.yaml", params\)\);/)
    expect(yaml.source).toMatch(/module\.exports = render;/)
    expect(yaml.source).toMatchSnapshot()
  })
})
