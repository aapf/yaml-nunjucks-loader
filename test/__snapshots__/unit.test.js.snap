// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[unit] loader works with loader options 1`] = `
"var YAML = require(\\"yaml\\");
var nunjucks = require(\\"nunjucks/browser/nunjucks-slim\\");
var env = new nunjucks.Environment(new nunjucks.PrecompiledLoader({
  \\"sample2.yaml\\": (function () { function root(env, context, frame, runtime, cb) {
var lineno = 0;
var colno = 0;
var output = \\"\\";
try {
var parentTemplate = null;
output += \\"resultItems:\\\\n  \\";
frame = frame.push();
var t_3 = runtime.contextOrFrameLookup(context, frame, \\"items\\");
if(t_3) {t_3 = runtime.fromIterator(t_3);
var t_2 = t_3.length;
for(var t_1=0; t_1 < t_3.length; t_1++) {
var t_4 = t_3[t_1];
frame.set(\\"item\\", t_4);
frame.set(\\"loop.index\\", t_1 + 1);
frame.set(\\"loop.index0\\", t_1);
frame.set(\\"loop.revindex\\", t_2 - t_1);
frame.set(\\"loop.revindex0\\", t_2 - t_1 - 1);
frame.set(\\"loop.first\\", t_1 === 0);
frame.set(\\"loop.last\\", t_1 === t_2 - 1);
frame.set(\\"loop.length\\", t_2);
output += \\"\\\\n  - name: \\";
output += runtime.suppressValue(runtime.memberLookup((t_4),\\"name\\"), env.opts.autoescape);
output += \\"\\\\n    value: \\";
output += runtime.suppressValue(runtime.memberLookup((t_4),\\"value\\"), env.opts.autoescape);
output += \\"\\\\n  \\";
;
}
}
frame = frame.pop();
output += \\"\\\\n\\";
if(parentTemplate) {
parentTemplate.rootRenderFunc(env, context, frame, runtime, cb);
} else {
cb(null, output);
}
;
} catch (e) {
  cb(runtime.handleError(e, lineno, colno));
}
}
return {
root: root
};
 })()
}), {\\"autoescape\\":true,\\"throwOnUndefined\\":true});
function render(params) {
  return YAML.parse(env.render(\\"sample2.yaml\\", params));
}
module.exports = render;"
`;

exports[`[unit] loader works with sample1.js 1`] = `
"var YAML = require(\\"yaml\\");
var nunjucks = require(\\"nunjucks/browser/nunjucks-slim\\");
var env = new nunjucks.Environment(new nunjucks.PrecompiledLoader({
  \\"sample1.yaml\\": (function () { function root(env, context, frame, runtime, cb) {
var lineno = 0;
var colno = 0;
var output = \\"\\";
try {
var parentTemplate = null;
output += \\"test:\\\\n  result: \\";
output += runtime.suppressValue(runtime.contextOrFrameLookup(context, frame, \\"name\\"), env.opts.autoescape);
output += \\"\\\\n\\";
if(parentTemplate) {
parentTemplate.rootRenderFunc(env, context, frame, runtime, cb);
} else {
cb(null, output);
}
;
} catch (e) {
  cb(runtime.handleError(e, lineno, colno));
}
}
return {
root: root
};
 })()
}), {\\"autoescape\\":false});
function render(params) {
  return YAML.parse(env.render(\\"sample1.yaml\\", params));
}
module.exports = render;"
`;

exports[`[unit] loader works with sample2.js 1`] = `
"var YAML = require(\\"yaml\\");
var nunjucks = require(\\"nunjucks/browser/nunjucks-slim\\");
var env = new nunjucks.Environment(new nunjucks.PrecompiledLoader({
  \\"sample2.yaml\\": (function () { function root(env, context, frame, runtime, cb) {
var lineno = 0;
var colno = 0;
var output = \\"\\";
try {
var parentTemplate = null;
output += \\"resultItems:\\\\n  \\";
frame = frame.push();
var t_3 = runtime.contextOrFrameLookup(context, frame, \\"items\\");
if(t_3) {t_3 = runtime.fromIterator(t_3);
var t_2 = t_3.length;
for(var t_1=0; t_1 < t_3.length; t_1++) {
var t_4 = t_3[t_1];
frame.set(\\"item\\", t_4);
frame.set(\\"loop.index\\", t_1 + 1);
frame.set(\\"loop.index0\\", t_1);
frame.set(\\"loop.revindex\\", t_2 - t_1);
frame.set(\\"loop.revindex0\\", t_2 - t_1 - 1);
frame.set(\\"loop.first\\", t_1 === 0);
frame.set(\\"loop.last\\", t_1 === t_2 - 1);
frame.set(\\"loop.length\\", t_2);
output += \\"\\\\n  - name: \\";
output += runtime.suppressValue(runtime.memberLookup((t_4),\\"name\\"), env.opts.autoescape);
output += \\"\\\\n    value: \\";
output += runtime.suppressValue(runtime.memberLookup((t_4),\\"value\\"), env.opts.autoescape);
output += \\"\\\\n  \\";
;
}
}
frame = frame.pop();
output += \\"\\\\n\\";
if(parentTemplate) {
parentTemplate.rootRenderFunc(env, context, frame, runtime, cb);
} else {
cb(null, output);
}
;
} catch (e) {
  cb(runtime.handleError(e, lineno, colno));
}
}
return {
root: root
};
 })()
}), {\\"autoescape\\":false});
function render(params) {
  return YAML.parse(env.render(\\"sample2.yaml\\", params));
}
module.exports = render;"
`;
